pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -120,9 +120,7 @@ def synchronize(remote_pk, repository_pk):
                 kickstart["repositories"].update({repodata: str(repository_pk)})
                 continue
             name = f"{repodata}-{kickstart['hash']}"
-            new_repository, created = RpmRepository.objects.get_or_create(
```

Comment:
I think we still want to keep track of which repos were created by the plugin and which were created by the user.  I'm not sure what the best solution to that is, whether we should have a different repository type for them, or if we should re-add a ```plugin-managed```-like field to ```RpmRepository```, or what.

Reply:
@fabricio-aguiar @dralley  we need to add this boolean field onto RpmRepository model,  i suggest calling it `sub_repo, default False. Port the logic that was removed from core into the plugin.
i.e.
1) filter the kickstart repo from CreatedResourcehttps://github.com/pulp/pulpcore/pull/375/files#diff-ae59196c41f139ec2b7e3ae3aa581a45L86
2) exclude from viewset https://github.com/pulp/pulpcore/pull/375/files#diff-03cc41a4bcada47c767bb1615dc53cefL54

In case in the the future we'd  change how we treat kickstart repos, we'd just always have sub_repo set to False.

Reply:
@daviddavis @bmbouter  in order to filter out the kickstart repo from CreatedResource like it was done in core https://github.com/pulp/pulpcore/pull/375/files#diff-ae59196c41f139ec2b7e3ae3aa581a45L86 we will need to define our own new_version() method in rpm plugin. any objections, concerns or other possible options?

Reply:
An alternative to adding a field to `RpmRepository` is to create a new typed repository like `DistributionTreeRepository`.

Reply:
I can see why you would need to keep doing this. For now I think the easiest thing to do is to define the `new_version()` as you suggest. In the future core could offer an optional flag that could disable it's creation of the CreatedResource. If someone filed an issue with that against core I would +1 it for 3.1